tests/test_arraycreation.py

```diff
@@ -716,3 +701,85 @@ def test_set_shape(shape):
     da.shape = shape
 
     assert_array_equal(na, da)
+
+
+def test_linspace_retstep_error():
+    with pytest.raises(NotImplementedError):
+        dpnp.linspace(2, 5, 3, retstep=True)
+
+
+def test_geomspace_zero_error():
+    with pytest.raises(ValueError):
+        dpnp.geomspace(0, 5, 3)
+        dpnp.geomspace(2, 0, 3)
+        dpnp.geomspace(0, 0, 3)
+
+
+def test_space_num_error():
+    with pytest.raises(ValueError):
+        dpnp.linspace(2, 5, -3)
+        dpnp.geomspace(2, 5, -3)
+        dpnp.logspace(2, 5, -3)
+        dpnp.linspace([2, 3], 5, -3)
+        dpnp.geomspace([2, 3], 5, -3)
+        dpnp.logspace([2, 3], 5, -3)
+
+
+@pytest.mark.parametrize("sign", [-1, 1])
+@pytest.mark.parametrize("dtype", get_all_dtypes())
+@pytest.mark.parametrize("num", [2, 4, 8, 3, 9, 27])
+@pytest.mark.parametrize("endpoint", [True, False])
+def test_geomspace(sign, dtype, num, endpoint):
+    start = 2 * sign
+    stop = 256 * sign
+
+    func = lambda xp: xp.geomspace(
+        start, stop, num, endpoint=endpoint, dtype=dtype
+    )
+
+    np_res = func(numpy)
+    dpnp_res = func(dpnp)
+
+    assert_allclose(dpnp_res, np_res)
+
+
+@pytest.mark.usefixtures("allow_fall_back_on_numpy")
+# dpnp.sign raise numpy fall back for complex dtype
+def test_geomspace_complex():
+    func = lambda xp: xp.geomspace(1j, 10j, num=10)
+    assert_allclose(func(numpy), func(dpnp))
+
+
+@pytest.mark.parametrize("axis", [0, 1])
+def test_geomspace_axis(axis):
+    func = lambda xp: xp.geomspace([2, 3], [20, 15], num=10, axis=axis)
+    assert_allclose(func(numpy), func(dpnp))
+
+
+@pytest.mark.parametrize("dtype", get_all_dtypes())
+@pytest.mark.parametrize("num", [2, 4, 8, 3, 9, 27])
+@pytest.mark.parametrize("base", [-2, 2])
+@pytest.mark.parametrize("endpoint", [True, False])
+def test_logspace(dtype, num, endpoint, base):
+    start = 2
+    stop = 5
+
+    func = lambda xp: xp.logspace(
+        start, stop, num, endpoint=endpoint, dtype=dtype, base=base
+    )
+
+    np_res = func(numpy)
+    dpnp_res = func(dpnp)
+
+    if dtype in [numpy.int64, numpy.int32]:
+        assert_allclose(dpnp_res, np_res, rtol=1)
+    else:
+        assert_allclose(dpnp_res, np_res)
+
+
+@pytest.mark.parametrize("axis", [0, 1])
+def test_logspace_axis(axis):
+    func = lambda xp: xp.logspace(
+        [2, 3], [20, 15], num=2, base=[[1, 3], [5, 7]], axis=axis
+    )
+    assert_allclose(func(numpy), func(dpnp))
```

Comment:
default dtype of result array is different on Iris XE, since no float64 there, need to use either a helper function:
```suggestion
    assert_dtype_allclose(func(dpnp), func(numpy))
```
or to pass dtype explicitly into `logspace` function by agrument.